ingestion/src/kldata.rs

```diff
@@ -35,47 +44,84 @@ struct ObsinnId {
     sensor_and_level: Option<(i32, i32)>,
 }
 
-fn parse_meta_field<T: FromStr>(field: &str, expected_key: &'static str) -> Result<T, Error>
-where
-    <T as FromStr>::Err: Debug,
-{
-    let (key, value) = field.split_once('=').ok_or_else(|| {
-        Error::Parse(format!(
-            "unexpected field in kldata header format: {}",
-            field
-        ))
-    })?;
+// TODO: maybe this can be a field in ObsinnChunk?
+#[derive(Default)]
+struct ObsinnHeader {
+    station_id: Option<i32>,
+    type_id: Option<i32>,
+    message_id: usize,
+    // TODO: we can parse it to Datatime if we decide we are going to use it
+    received_time: Option<String>,
```

Comment:
I don't think these fields should be `Option`, since the only place they can be `None` is inside the constructor. I think it would be better to remove the `Option` here and use separate variables (which themselves are `Option`) in the contructor, and build the struct like:

```
return ObsinnHeader {
    station_id: station_id.ok_or(...)?,
    ...
```

Reply:
Yep, I thought about that but then forgot to change them. I'm only keeping `received_time` as on `Option` since the message is still valid if that field is missing